e2e/tests/test_e2e.py

```diff
@@ -694,13 +728,23 @@ def get_service_type(self, svc_labels, namespace='default'):
     def check_service_annotations(self, svc_labels, annotations, namespace='default'):
         svcs = self.api.core_v1.list_namespaced_service(namespace, label_selector=svc_labels, limit=1).items
         for svc in svcs:
-            if len(svc.metadata.annotations) != len(annotations):
+            if len(svc.metadata.annotations) < len(annotations):
                 return False
             for key in svc.metadata.annotations:
                 if svc.metadata.annotations[key] != annotations[key]:
                     return False
         return True
 
+    def check_statefulset_annotations(self, sset_labels, annotations, namespace='default'):
+        ssets = self.api.apps_v1.list_namespaced_stateful_set(namespace, label_selector=sset_labels, limit=1).items
+        for sset in ssets:
+            if len(sset.metadata.annotations) < len(annotations):
```

Comment:
theses lines can be imo written slightly clearer:
```python
for key, value in annotations:
  if key not in sset.metadata.annotations or sset.metadata.annotations[key] != value:
    return False
```
the first condition avoids the `KeyError` if annotations indeed has not been propagated to the stateful set